pkg/apis/core/helper/shoot_test.go

```diff
@@ -616,4 +617,52 @@ var _ = Describe("Helper", func() {
 		Entry("with KubeProxy in IPVS mode", &core.KubeProxyConfig{Enabled: ptr.To(true), Mode: ptr.To(core.ProxyModeIPVS)}, true),
 		Entry("with KubeProxy in IPTables mode", &core.KubeProxyConfig{Enabled: ptr.To(true), Mode: ptr.To(core.ProxyModeIPTables)}, false),
 	)
+
+	Describe("#IsShootAutonomous", func() {
+		It("should return true (single worker pool with control plane configuration)", func() {
+			shoot := &core.Shoot{Spec: core.ShootSpec{Provider: core.Provider{Workers: []core.Worker{
+				{ControlPlane: &core.WorkerControlPlane{}},
+			}}}}
+			Expect(IsShootAutonomous(shoot.Spec.Provider.Workers)).To(BeTrue())
+		})
+
+		It("should return true (multiple worker pools, one with control plane configuration)", func() {
+			shoot := &core.Shoot{Spec: core.ShootSpec{Provider: core.Provider{Workers: []core.Worker{
+				{},
+				{ControlPlane: &core.WorkerControlPlane{}},
+				{},
+			}}}}
+			Expect(IsShootAutonomous(shoot.Spec.Provider.Workers)).To(BeTrue())
+		})
+
+		It("should return false (no worker pools)", func() {
+			shoot := &core.Shoot{}
+			Expect(IsShootAutonomous(shoot.Spec.Provider.Workers)).To(BeFalse())
+		})
+
+		It("should return false (worker pools, but none with control plane configuration)", func() {
+			shoot := &core.Shoot{Spec: core.ShootSpec{Provider: core.Provider{Workers: []core.Worker{
+				{},
+				{},
+				{},
+			}}}}
+			Expect(IsShootAutonomous(shoot.Spec.Provider.Workers)).To(BeFalse())
+		})
+	})
+
+	Describe("#ControlPlaneWorkerPoolForShoot", func() {
+		It("should return nil because shoot is not autonomous", func() {
+			shoot := &core.Shoot{}
```

Comment:
To also test worker pools without `ControlPlane` should not be considered.
```suggestion
			shoot := &core.Shoot{Spec: core.ShootSpec{Provider: core.Provider{Workers: []core.Worker{{}}}}
```